contracts/IssuingERC20.sol

```diff
@@ -0,0 +1,157 @@
+        return _symbol;
+    }
+
+    function decimals() public view returns (uint8) {
+        return _decimals;
+    }
+
+    function backing() public view returns (address) {
+        return _backing;
+    }
+
+    function source() public view returns (address) {
+        return _source;
+    }
+
+    function totalSupply() public view override returns (uint256) {
+        return _totalSupply;
+    }
+
+    function balanceOf(address account) public view override returns (uint256) {
+        return _balances[account];
+    }
+
+    function transfer(address recipient, uint256 amount) public virtual override returns (bool) {
+        _transfer(msg.sender, recipient, amount);
+        return true;
+    }
+
+    function allowance(address owner, address spender) public view virtual override returns (uint256) {
+        return _allowances[owner][spender];
+    }
+
+    function approve(address spender, uint256 amount) public virtual override returns (bool) {
+        _approve(msg.sender, spender, amount);
+        return true;
+    }
+
+    function transferFrom(address sender, address recipient, uint256 amount) public virtual override returns (bool) {
+        _transfer(sender, recipient, amount);
+        _approve(sender, msg.sender, _allowances[sender][msg.sender].sub(amount, "ERC20: transfer amount exceeds allowance"));
+        return true;
+    }
+
+    function increaseAllowance(address spender, uint256 addedValue) public virtual returns (bool) {
+        _approve(msg.sender, spender, _allowances[msg.sender][spender].add(addedValue));
+        return true;
+    }
+
+    function decreaseAllowance(address spender, uint256 subtractedValue) public virtual returns (bool) {
+        _approve(msg.sender, spender, _allowances[msg.sender][spender].sub(subtractedValue, "ERC20: decreased allowance below zero"));
+        return true;
+    }
+
+    function _transfer(address sender, address recipient, uint256 amount) internal virtual {
+        require(sender != address(0), "ERC20: transfer from the zero address");
+        require(recipient != address(0), "ERC20: transfer to the zero address");
+
+        _beforeTokenTransfer(sender, recipient, amount);
+
+        _balances[sender] = _balances[sender].sub(amount, "ERC20: transfer amount exceeds balance");
+        _balances[recipient] = _balances[recipient].add(amount);
+        emit Transfer(sender, recipient, amount);
+    }
+
+    function mint(address account, uint256 amount) external onlyOwner {
+        _mint(account, amount);
+    }
+
+    function transferToSource(address recipient, uint256 amount, bool is_native) external {
+        require(amount > 0, "can not transfer amount zero");
+        _burn(msg.sender, amount);
+        (bool success, ) = BURN_PRECOMPILE.call(abi.encode(_backing, _source, recipient, amount, is_native));
+        require(success, "burn: call burn precompile failed");
+        emit Burn(recipient, amount);
+    }
```

Comment:
Not so sure that we should put `backing` and `source` and `transferToSource` inside the ERC20.

Since ERC20 address is registered in third parties such as wallet and explorer, and we might want the modules related to backing(lock/unlock) and issue/redeem being located in separate contracts, which can be upgradable by the system.

Reply:
yes, before I think it'll be good to call erc20 contract directly by issuing pallet. So I put some options into ERC20.
For keeping it pure like other erc20s. It maybe better to move this mapping relationship to the factory contract.